src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -438,10 +440,56 @@ public async Task<List<Stack>> GetCloudFormationStacks()
             return await cloudFormationClient.Paginators.DescribeStacks(new DescribeStacksRequest()).Stacks.ToListAsync();
         }
 
-        public async Task<GetCallerIdentityResponse> GetCallerIdentity()
+        public async Task<GetCallerIdentityResponse> GetCallerIdentity(AWSCredentials awsCredentials, string awsRegion)
         {
-            using var stsClient = _awsClientFactory.GetAWSClient<IAmazonSecurityTokenService>();
-            return await stsClient.GetCallerIdentityAsync(new GetCallerIdentityRequest());
+            var request = new GetCallerIdentityRequest();
+
+            try
+            {
+                _awsClientFactory.ConfigureAWSOptions(awsOption =>
+                {
+                    awsOption.Credentials = awsCredentials;
+                    awsOption.Region = RegionEndpoint.GetBySystemName(awsRegion);
+                });
+
+                using var stsClient = _awsClientFactory.GetAWSClient<IAmazonSecurityTokenService>();
+                return await stsClient.GetCallerIdentityAsync(request);
+            }
+            catch (Exception ex)
+            {
+                var regionEndpoint = RegionEndpoint.GetBySystemName(awsRegion);
+                if (regionEndpoint.PartitionName.Equals("aws") && !awsRegion.Equals(Constants.CLI.DEFAULT_AWS_REGION))
+                {
+                    try
+                    {
+                        _awsClientFactory.ConfigureAWSOptions(awsOption =>
+                        {
+                            awsOption.Credentials = awsCredentials;
+                            awsOption.Region = RegionEndpoint.GetBySystemName(Constants.CLI.DEFAULT_AWS_REGION);
+                        });
+
+                        using var stsClient = _awsClientFactory.GetAWSClient<IAmazonSecurityTokenService>();
+                        await stsClient.GetCallerIdentityAsync(request);
+                    }
+                    catch (Exception e)
+                    {
+                        throw new UnableToAccessAWSRegionException(
+                           DeployToolErrorCode.UnableToAccessAWSRegion,
+                           $"We were unable to access the AWS region '{awsRegion}'. Make sure you have correct permissions for that region and the region is accessible.",
+                           e);
+                    }
+
+                    throw new UnableToAccessAWSRegionException(
```

Comment:
Did you mean to return on line 472? This appears to be unconditionally thrown.
If that is the case, please add a test that verifies that line 472 returns as expected.

Reply:
No this is intentional. Any time you step into the catch block, and exception will be thrown. It's just a matter of which exception and error code needs to be thrown. I have added a unit test to cover that case.

Reply:
(non blocking, at your discretion) Consider moving this whole catch block logic (line 451) to a separate method to help make it easier to understand the intent.